sys/vfs/vfs.c

```diff
@@ -450,10 +491,16 @@ int vfs_mount(vfs_mount_t *mountp)
 int vfs_umount(vfs_mount_t *mountp)
 {
     DEBUG("vfs_umount: %p\n", (void *)mountp);
-    if ((mountp == NULL) || (mountp->mount_point == NULL)) {
+    int ret = check_mount(mountp);
+    switch (ret) {
+    case 0:
+        DEBUG("vfs_umount: not mounted\n");
+        mutex_unlock(&_mount_mutex);
         return -EINVAL;
+    case -EINVAL:
+        DEBUG("vfs_umount: invalid fs\n");
+        return ret;
```

Comment:
What I meant with default case was to add:
```c
default:
return -EINVAL;
```
to have as fallback if check_mount misbehaves.

Reply:
See last commit